File: src/php/core/src/Style/MetaSource.php
<?php

declare(strict_types=1);

namespace DiceBear\Style;

/**
 * Read-only view over the `meta.source` block of a style definition.
 */
class MetaSource
{
    /**
     * @param array<string, mixed> $data
     */
    public function __construct(private readonly array $data) {}

    /**
     * Returns the source name (e.g. the original work title), or `null` when
     * not set.
     */
    public function name(): ?string
    {
        return $this->data['name'] ?? null;
    }

    /**
     * Returns the URL of the source, or `null` when not set.
     */
    public function url(): ?string
    {
        return $this->data['url'] ?? null;
    }
}

File: src/rust/core/src/style/meta_source.rs
//! Read-only view over a style definition's `meta.source` block.

use serde::Deserialize;

#[derive(Deserialize, Default)]
pub(crate) struct MetaSource {
    name: Option<String>,
    url: Option<String>,
}

impl MetaSource {
    pub(crate) fn name(&self) -> Option<&str> {
        self.name.as_deref()
    }

    pub(crate) fn url(&self) -> Option<&str> {
        self.url.as_deref()
    }
}
